src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -452,13 +451,18 @@ public void ResetOptions()
                     payload.RunSettings));
         }
 
-        // TODO (copoiena): Collect metrics ?
+        var runConfiguration = XmlRunSettingsUtilities.GetRunConfigurationNode(payload.RunSettings);
+        var requestData = GetRequestData(protocolConfig);
+
+        // Collect metrics & commands.
+        CollectMetrics(requestData, runConfiguration);
+        LogCommandsTelemetryPoints(requestData);
 
         lock (_syncObject)
         {
             try
             {
-                EqtTrace.Info("TestRequestManager.StartTestRunner: Synchronization context taken.");
+                EqtTrace.Info("TestRequestManager.StartTestSession: Synchronization context taken.");
```

Comment:
I know we are pretty strict about backward compatibility. Do we consider that our "logs" are part of the API or not?

Reply:
Nope, we're updating to be able to find in a fast way during analysis. EqtTrace won't be shown to the user.

Reply:
It's not written in a log or whatever? I mean we are not expecting users to have written something that would parse logs to look for something (or at least if they do, we "don't care" because that's not part of the API, right?)

Reply:
I agree, the logs aren't part of the API and I don't think we should treat them as such either going forward. The logs are a tool for us to find issues in TP, users can parse the console output and TRX but we should draw the line at logs (and to be completely honest TRX parsing is causing us a lot of headaches too but it's too late to change that now :D).

This approach (i.e. treating logs as API) would mostly hurt us because we won't be able to change them unless we add fresh logs while the benefit to the user is not immediately apparent. We haven't heard of anyone parsing the logs yet and even if they do I think we shouldn't support that behavior.